### ISSUE TITLE
[jsonrpc] Request Id returns null when request errs. 

### ISSUE THREAD

**Issue by bubunyo** [go-kit/kit#996]
When a request with an id is successful the request-id is returned successfully in the response. When the request errs, the request-id returned is null instead of the original request id which was submitted as part of the request. 

Sample Request
```
{
    "id": 1,
    "method": "Create",
    "params": {
        "name": "Master Garzy",
        "email": 4    <----- a string is expected here
    }
}
```
Response
```
{
    "jsonrpc": "2.0",
    "error": {
        "code": -32000,
        "message": "Create - Bad Request: json: cannot unmarshal number into Go struct field CreateRequest.email of type string"
    },
    "id": null <----- Null Value instead of 1 from the request above. 
}
```

**Comment by bubunyo** [comment]
I can work on a PR this week.

**Comment by bubunyo** [comment]
[transport/http/jsonrpc/server_test.go L112](https://github.com/go-kit/kit/blob/556100560949062d23fe05d9fda4ce173c30c59f/transport/http/jsonrpc/server_test.go#L112) is a test that replicates this issue and it passes. I am going to do a little bit more digging on my end to see what the issue is.

**Comment by bubunyo** [comment]
I think this issue has been solved by #959. I will try updating my gokit version to see if that fixes it. For now, I am closing.
